examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -95,23 +97,46 @@ EmberAfStatus OTAProviderExample::HandleQueryImage(chip::app::CommandHandler * c
                                                    const chip::app::ConcreteCommandPath & commandPath,
                                                    const QueryImage::DecodableType & commandData)
 {
-    // TODO: add confiuration for returning BUSY status
+    ChipLogDetail(SoftwareUpdate, "Requestor endpoint:%" PRIu16 " node-id:0x" ChipLogFormatX64, commandPath.mEndpointId,
+                  ChipLogValueX64(commandObj->SourceNodeId()));
 
-    OTAQueryStatus queryStatus  = OTAQueryStatus::kNotAvailable;
     uint32_t newSoftwareVersion = commandData.softwareVersion + 1; // This implementation will always indicate that an update is
                                                                    // available (if the user provides a file).
     constexpr char kExampleSoftwareString[] = "Example-Image-V0.1";
-    bool userConsentNeeded                  = false;
     uint8_t updateToken[kUpdateTokenLen]    = { 0 };
     char strBuf[kUpdateTokenStrLen]         = { 0 };
     char uriBuf[kUriMaxLen]                 = { 0 };
+    QueryImageResponse::Type response;
+
+    OTAQueryStatus queryStatus = (strlen(mOTAFilePath) != 0) ? OTAQueryStatus::kUpdateAvailable : OTAQueryStatus::kNotAvailable;
+    bool requestorCanConsent   = commandData.requestorCanConsent.ValueOr(false);
+
+    if (queryStatus == OTAQueryStatus::kUpdateAvailable && mUserConsentDelegate != nullptr)
+    {
+        UserConsentState state = mUserConsentDelegate->GetUserConsentState(commandObj->SourceNodeId(), commandPath.mEndpointId,
```

Comment:
What if one of the platforms implements `GetUserConsentState()` to do some blocking work (for example, waiting for a user to press a button on m5stack)? I don't think it would be good to have this function blocked for an indefinite amount of time.

Seems like this should be designed to be more asynchronous and not expect an answer right away.

Reply:
PTAL these https://github.com/project-chip/connectedhomeip/pull/13816#discussion_r789738659 and https://github.com/project-chip/connectedhomeip/pull/13816#pullrequestreview-859664595 from @tcarmelveilleux.

I think the value returned by `GetUserConsentState()` can be helpful in both async and sync calls. `kGranted` and `kDenied` are for the deferred user consents. `kObtaining` should be returned when action is async, so that on the next query the same call can return `kGranted` or `kDenied`.

In previous implementation I had the OTA provider deal with the user consent states for handing async API() response but, IMO delegating that work to platform looks clean.

Reply:
I think the important part Tennessee highlighted is 
> Relaying of a previously stored consent signal, previously provided by a User at some point in the past
Works for me!